cloud/openstack/os_server.py

```diff
@@ -489,14 +489,18 @@ def _delete_floating_ip_list(cloud, server, extra_ips):
             server=server.id, address=ip)
 
 
-def _check_floating_ips(module, cloud, server):
+def _check_external_ips(module, cloud, server):
     changed = False
 
     auto_ip = module.params['auto_ip']
     floating_ips = module.params['floating_ips']
     floating_ip_pools = module.params['floating_ip_pools']
 
-    if floating_ip_pools or floating_ips or auto_ip:
+    if auto_ip:
+        if not cloud.get_server_public_ip(server):
+            server = cloud._add_auto_ip(
+                server, wait=wait, timeout=timeout, reuse=reuse)
```

Comment:
I don't possibly see how your posted test case is working since 'reuse' is not a defined variable here. Are you certain this code is getting hit by your test case?